### PR TITLE
Convert tests to jest

### PR DESCRIPTION
**Pull request by kenr**
* Change tests to be ran by jest with babel transpilation, fulfilling NodeJS version requirement (>=6.5.7)
* Change and move dfu test that require human intervention into examples
* Minor bug fixes
* Add support for filtering devices to use based on device family and blacklist

This PR requires [nrf-device-lister](https://github.com/NordicSemiconductor/nrf-device-lister-js) to add  support for NodeJS 6.x (missing padStart polyfill).

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (IvanSanchez) [inline .babelrc]: Suggestion: put the contents of this file inside a `babel` field inside `package.json` (as explained in https://babeljs.io/docs/usage/babelrc/ ). That way prevents cluttering the root of the repo with small files.
This is just a personal preference, though.

File: examples/dfu.js [hunk 2]
```diff
@@ -97,17 +97,84 @@ function performDfu(adapter, targetAddress, pathToZip) {
...
+    const adapter = adapterFactory.createAdapter(apiVersion, port, '');
+
+    openAdapter(adapter).then(() => {
+        console.log(`Starting DFU process towards ${targetAddress}`);
+        return performDfu(adapter, targetAddress, pathToDfuZip);
+    }).catch(error => {
+        console.log(error);
+        process.exit(-1);
     });
+}
```
Comment (mrodem) [inline examples/dfu.js]: Could perhaps remove the DFU zip and hex files now? I believe they were only used by the old DFU jest test that has recently been removed.

File: test/connection.test.js [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+        });
+    });
+}
+
+describe('the API', async () => {
+    let centralAdapter;
+    let peripheralAdapter;
+
+    beforeAll(async () => {
+        [centralAdapter, peripheralAdapter] = await Promise.all([grabAdapter(), grabAdapter()]);
```
Comment (mrodem) [inline test/connection.test.js]: Have you tested this with two J-Link devices? A potential issue with multiple `grabAdapter()` calls in parallel could be that multiple pc-nrfjprog-js operations are run at the same time, which may cause crash.

File: .babelrc (hunk 1 above)
Comment (kenr) [reply .babelrc]: Ok. Moving babel config into package.json.

File: examples/heart_rate_collector.js [hunk 4]
```diff
@@ -373,10 +373,11 @@ if (process.argv.length !== 4) {
     }
 
     if (!apiVersion) {
-        if (!['v2', 'v3'].includes(apiVersion)) {
-            console.error(`SD_API_VERSION must be v2 or v3, argument provided is ${apiVersion}`);
-            process.exit(-1);
-        }
+        console.error('SD_API_VERSION must be provided');
```
[the remark this answers is not in the document]
Comment (kenr) [reply examples/heart_rate_collector.js]: Yes. Adding more checks.

File: examples/dfu.js (hunk 2 above)
Comment (kenr) [reply examples/dfu.js]: I think those can be used by the example code. That way the user do not have to create those to run the example.

File: test/connection.test.js (hunk 3 above)
Comment (kenr) [reply test/connection.test.js]: Yes, it has been tested with two J-Link devices and I've not had any trouble with it.

**Pull request merged**